tests/test_cgi.py

```diff
@@ -68,9 +71,9 @@ def start_response(status, headers):
             test_status[0] = status
             test_headers[0] = headers
         for str_data in application(environ, start_response):
-            if not isinstance(str_data, str):
+            if not isinstance(str_data, bytes):
                 err = "answer is not a byte string: %r" % limit(str_data, 30)
                 self.assertTrue(False, err)
-            test_response += str_data
+            test_response += str(str_data)
```

Comment:
`str(bytes)` does the wrong thing on Python 3:

```
>>> str(b'hello')
"b'hello'"
```

While this test only looks for a substring and doesn't care about the extra b prefix and extra quotes, I'd prefer if we did it correctly.

(Which means `test_response = b""` and concatenating it all as bytes before calling `test_response.decode("UTF-8")`, since in theory the response may be split in the middle of a multibyte character.)

Reply:
Have implemented with `self.assertTrue(b"Generated by LinkChecker" in test_response)`. Only because decoding made the line too long. test_response isn't used subsequently.